src/basic/efivars.c

```diff
@@ -365,10 +352,42 @@ int systemd_efi_options_variable(char **line) {
                 return -EPERM;
         }
 
-        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", line);
+        r = efi_get_variable_string(EFI_VENDOR_SYSTEMD, "SystemdOptions", &line);
         if (r == -ENOENT)
                 return -ENODATA;
+        if (r < 0)
+                return r;
 
+        r = mkdir_safe_label("/run/systemd/efivars", 0700, 0, 0, MKDIR_WARN_MODE);
```

Comment:
I'd just use 0755 here, if we add more stuff there and it shall be inaccessible it should be the individual files that protect access I think

Reply:
Done (actually asking `write_string_file()` to create the directory now.)